contracts/StorageAccessible.sol

```diff
@@ -52,20 +52,47 @@ contract StorageAccessible {
      */
     function simulateDelegatecall(
         address targetContract,
-        bytes memory calldataPayload
+        bytes calldata calldataPayload
     ) public returns (bytes memory response) {
-        bytes memory innerCall = abi.encodeWithSelector(
-            this.simulateDelegatecallInternal.selector,
-            targetContract,
-            calldataPayload
-        );
-        (, response) = address(this).call(innerCall);
-        bool innerSuccess = response[response.length - 1] == 0x01;
-        setLength(response, response.length - 1);
-        if (innerSuccess) {
-            return response;
-        } else {
-            revertWith(response);
+        bytes4 selector = this.simulateDelegatecallInternal.selector;
```

Comment:
would it make sense to have this in assembly too, to basically fully avoid solidity inside the function. Not sure if this assignment to `bytes4` generates opcodes for masking.

Reply:
The `.selector` notation is not available in inline assembly :sob:. I can just use a hardcoded value.

Reply:
Yeah that was what I meant (to hardcode the value)